doc/Language/syntax.pod

```diff
@@ -341,22 +342,67 @@ arrays themselves:
     # does not flatten:
     say [[@a], [3, 4]].elems;   # 2
 
-=head3 Hash literals
+=head3 L<Hash|/type/Hash> literals
 
-=comment TODO
+A pair of curly braces can surround a list of pairs to form a
+L<Hash|/type/Hash> literal; typically there is a comma-delimited list
+of pairs inside:
+
+    say {a => 73, foo => "fish"}.keys.join(" ");   # a foo
+    #   ^^^^^^^^^^^^^^^^^^^^^^^^ Hash constructor
+
+Most often this is used with arrow pairs.
 
-=head3 Regex literals
+=head3 L<Regex|/type/Regex> literals
 
 =comment TODO
 
+A regex is declared with slashes like C</foo/>. Note that this C<//> syntax is shorthand for the full C<rx//> syntax.
+
+You can assign regexes to variables, just like with other literals.
+
+    my $pattern = /foo/;
+    say "Match!" if "asdfoosdf" ~~ $pattern; # Match!
+
 =head2 Declarations
 
 =comment TODO
 
-=head2 Subroutine calls
+=head3 Variable declaration
+
+    my $x;                      # simple lexical variable
+    my $x = 7;                  # initialize the variable
+    my Int $x = 7;              # declare the type
+    my Int $x:D = 7;            # specify that the value must be defined (not undef)
+    my Int $x where * > 3 = 7;  # restrict the value based on a function
```

Comment:
s/restrict/constrain/ (just to be consistent with the .constraints method). Maybe s/function/block/.